pkg/operator/controller/controller_dns_configmap_test.go

```diff
@@ -336,3 +337,1449 @@ foo.com:5353 {
 		t.Errorf("unexpected Corefile; got:\n%s\nexpected:\n%s\n", cmToCheckTraceLogLevel.Data["Corefile"], expectedCorefileToCheckIfTraceLogLevelIsSet)
 	}
 }
+
+func TestDesiredDNSConfigmapUpstreamResolvers(t *testing.T) {
+	testCases := []struct {
+		name             string
+		dns              *operatorv1.DNS
+		expectedCoreFile string
+		expectedError    error
+	}{
+		{
+			name: "CR of 5 upstreams in upstreamResolvers should return a coreFile with 5 upstreams defined",
+			dns: &operatorv1.DNS{
+				ObjectMeta: metav1.ObjectMeta{
+					Name: DefaultDNSController,
+				},
+				Spec: operatorv1.DNSSpec{
+					UpstreamResolvers: operatorv1.UpstreamResolvers{
+						//{"1.2.3.4", "9.8.7.6", "6.4.3.2:53", "2.3.4.5:5353", "127.0.0.53"},
+						Upstreams: []operatorv1.Upstream{
+							{
+								Type:    operatorv1.NetworkResolverType,
+								Address: "1.2.3.4",
+							},
+							{
+								Type:    operatorv1.NetworkResolverType,
+								Address: "9.8.7.6",
+							},
+							{
+								Type:    operatorv1.NetworkResolverType,
+								Address: "6.4.3.2",
+								Port:    53,
+							},
+							{
+								Type:    operatorv1.NetworkResolverType,
+								Address: "2.3.4.5",
+								Port:    5353,
+							},
+							{
+								Type:    operatorv1.NetworkResolverType,
+								Address: "127.0.0.53",
+							},
+						},
+						Policy: operatorv1.RoundRobinForwardingPolicy,
+					},
+					Servers: []operatorv1.Server{
```

Comment:
By the way, if the goal of this test is the upstream resolvers, maybe it's not that necessary to add 4 servers?
Would only one be enough? That can save a lot of space.